internal/servers/repository.go

```diff
@@ -75,46 +82,114 @@ func (r *Repository) ListServers(ctx context.Context, serverType ServerType, opt
 	return servers, nil
 }
 
+// ServerTag holds the information for the server_tag table for Gorm.
+type ServerTag struct {
+	ServerId string
+	Key      string
+	Value    string
+}
+
+// TableName overrides the table name used by ServerTag to `server_tag`
+func (ServerTag) TableName() string {
+	return "server_tag"
+}
+
+// ListTagsForServers pulls out tag tuples into ServerTag structs for the
+// given server ID values.
+func (r *Repository) ListTagsForServers(ctx context.Context, serverIds []string, opt ...Option) ([]*ServerTag, error) {
+	var serverTags []*ServerTag
+	if err := r.reader.SearchWhere(
+		ctx,
+		&serverTags,
+		"server_id in (?)",
+		[]interface{}{serverIds},
+		db.WithLimit(-1),
+	); err != nil {
+		return nil, fmt.Errorf("error listing server tags with server IDs %#v: %w", serverIds, err)
```

Comment:
should this be using the errors pkg?

Reply:
I don't know? I know nothing about that package or how you use it.

Reply:
Done.